src/app/components/add-board-modal/add-board-modal.component.html

```diff
@@ -87,5 +87,5 @@ <h4>Set Background Image:</h4>
 </div>
 <div mat-dialog-actions>
   <button mat-button (click)="onNoClick()">Cancel</button>
-  <button mat-button  cdkFocusInitial [disabled]="boardNameInput.invalid || projectInput.invalid" (click) ="handleDialogSubmit()">Create Board!</button>
+  <button mat-button  cdkFocusInitial [disabled]="(taskMessage && !taskTitle.trim()) || (taskTitle && !taskTitle.trim()) || boardNameInput.invalid || projectInput.invalid " (click) ="handleDialogSubmit()">Create Board!</button>
```

Comment:
taskMessage is optional so !taskTitle.trim() should be enough. Nice that you added in the trim. I don't think the other fields do that so we should prob add that in the future and refactor how we do form validation.
```suggestion
  <button mat-button  cdkFocusInitial [disabled]="!taskTitle.trim() || boardNameInput.invalid || projectInput.invalid " (click) ="handleDialogSubmit()">Create Board!</button>
```

Reply:
If we just do !taskTitle.trim(), wouldn't that mean you need to have a task title at all times to update/create the board? 
The reason i added the trim was because if the user enters just whitespaces (and i don't check for whitespaces), when you try to see the view the task on the page, it would be blank. So by doing !taskTitle.trim(), it sees if the title was whitespace, if it was it wont let it update.

Reply:
in `add-board-modal.component.ts` taskTitle is defaulted to `''`. Task title is mandatory I believe, so you need to have it all times. You reasoning makes sense, just that taskMessage is optional so we don't have to check.

Reply:
oh ok, in the ss for the issue it showed 'No Task Created' so I thought having a task (also task title) was not mandatory at all times.

Reply:
@JoelWiebe are tasks mandatory for boards? If thats the case then no change needed

Reply:
@l3n0ire @LunarFang416 Tasks are not mandatory at this point; however, if a task message exists then a task title is required :)

Reply:
This does bring up an interesting point though, if tasks are not required should we disable the task icon rather than have the student open it up just to see a message that no task was created, haha?

Reply:
Sorry for the scope creep, but what do you think about us also removing the "View Full Task" button in the snackbar if the task can already fully fit in the snackbar?

Reply:
And finally... I see that in at least some of the branches, tasks are mandatory when creating a new board. I'll create a follow up PR to remove tasks as a requirement when creating a new board (to make board creation super quick and simple)!